contracts/PolygonTokenBridger.sol

```diff
@@ -74,12 +96,22 @@ contract PolygonTokenBridger is Lockable {
      * @notice Called by someone to send tokens to the destination, which should be set to the HubPool.
      * @param token Token to send to destination.
      */
-    function retrieve(IERC20 token) public nonReentrant {
+    function retrieve(IERC20 token) public nonReentrant onlyChainId(l1ChainId) {
         token.safeTransfer(destination, token.balanceOf(address(this)));
     }
 
     receive() external payable {
-        // Note: this should only happen on the mainnet side where ETH is sent to the contract directly by the bridge.
-        if (functionCallStackOriginatesFromOutsideThisContract()) l1Weth.deposit{ value: address(this).balance }();
+        if (functionCallStackOriginatesFromOutsideThisContract()) {
+            // This should only happen on the mainnet side where ETH is sent to the contract directly by the bridge.
+            _requireChainId(l1ChainId);
+            l1Weth.deposit{ value: address(this).balance }();
+        } else {
```

Comment:
more succinct: `_requireChainId(functionCallStackOriginatesFromOutsideThisContract() ? l1ChainId : l2ChainId)`

Reply:
Only reason I don't want to do this is that we still need to do the l1Weth.deposit call in a separate branch, which makes this feel a bit messier since we're doing the same branch twice.